File: backend/entity/payment_review.go
// entity/payment_review.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

// บันทึกผลการตรวจสลิป/การชำระ (โดยแอดมิน)
type PaymentReview struct {
	gorm.Model
	VerifiedAt time.Time `json:"verified_at"`
	Title      string    `json:"title"`
	Result     string    `json:"result"` // e.g. "approved" | "rejected"
	Note       string    `json:"note"`

	UserID uint  `json:"user_id"` // ผู้ตรวจ
	User   *User `gorm:"foreignKey:UserID" json:"user,omitempty"`

	// ถ้าต้องการโยงกับ Payment/Order เฉพาะเจาะจง สามารถเพิ่ม FK เพิ่มได้ภายหลัง
}

File: frontend/src/interfaces/PaymentReview.ts
import type { User } from "./User";

export interface PaymentReview {
  ID: number;
  verified_at: string; // ISO datetime
  title: string;
  result: string;      // "approved" | "rejected" | ...
  note: string;

  user_id: number;     // ผู้ตรวจ
  user?: User;
}

export interface CreatePaymentReviewRequest {
  verified_at?: string;
  title: string;
  result: string;
  note: string;
  user_id: number;
}
